File: 18015_18174_ProjetoED/TrensCidades/TrensCidades/CanvasView.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;

namespace TrensCidades
{
    //Gustavo Henrique de Meira - 18015
    //Pedro Gomes Moreira - 18174

    public class CanvasView : View
    {
        private const float mapaTotalX = 358.5f;
        private const float mapaTotalY = 289f;

        public CanvasView(Context context) : base(context)
        {
        }

        public CanvasView(Context context, IAttributeSet attrs) : base(context, attrs)
        {
        }

        public CanvasView(Context context, IAttributeSet attrs, int defStyleAttr) : base(context, attrs, defStyleAttr)
        {
        }

        public CanvasView(Context context, IAttributeSet attrs, int defStyleAttr, int defStyleRes) : base(context, attrs, defStyleAttr, defStyleRes)
        {
        }

        protected CanvasView(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
        {
        }

        protected override void OnDraw(Canvas canvas)
        {
            Bitmap mapa = BitmapFactory.DecodeResource(Resources, Resource.Drawable.Mapa);

            canvas.DrawBitmap(mapa,0, 0, null);

            base.OnDraw(canvas);
        }
    }
}

File: 18015_18174_ProjetoED/TrensCidadesAndroid/app/src/main/java/com/example/trenscidadesandroid/CanvasView.java
package com.example.trenscidadesandroid;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.Log;
import android.view.View;

//Gustavo Henrique de Meira - 18015
//Pedro Gomes Moreira - 18174

public class CanvasView extends View
{
    private static final float TOTAL_X = 358.5f;
    private static final float TOTAL_Y = 289f;
    private static final float ESPESSURA = 5.0f;
    private static final int COR_LINHA = Color.RED;

    private Paint paint; //propriedades do formato (cor)

    public CanvasView(
        Context context)
    {
        super(context);
    }

    @Override
    protected void onDraw(
        Canvas canvas)
    {
        super.onDraw(canvas);

        Log.d("MSG", "chegou");

        this.paint = new Paint();
        this.paint.setColor(COR_LINHA);
        this.paint.setStrokeWidth(ESPESSURA);

        //pega imagem põe no bitmap
        Bitmap imagem = BitmapFactory.decodeResource(getResources(), R.drawable.mapa);
        canvas.drawBitmap(imagem, 0, 0,null);


    }
}
